### PR TITLE
Extend tests in TFLite micro for conv and depth_conv to support dilat…

### PR DESCRIPTION
**Pull request by giorgio-arenarm**
…ion != 1

### CODE REVIEW

**Review by njeffrie — DISMISSED** [review]
You will probably have to resolve a few conflicts vs top-of-tree, since I submitted a change yesterday affecting these files.  Looks good otherwise.

**Comment by giorgio-arenarm** [comment]
Done rebasing, I just needed to add a couple of parameters indicating in/out zero_points when calling the PerChannel tests. It should be good to go now

File: tensorflow/lite/micro/kernels/depthwise_conv_test.cc [hunk 1]
```diff
@@ -27,12 +27,10 @@ static const int kMaxFilterChannels = 64;
...
-TfLiteStatus ValidateDepthwiseConvGoldens(TfLiteTensor* tensors,
-                                          int tensors_size,
-                                          const T* expected_output_data,
-                                          T* output_data, int output_length,
-                                          TfLiteFusedActivation activation,
-                                          float tolerance = 1e-5) {
+TfLiteStatus ValidateDepthwiseConvGoldens(
+    TfLiteTensor* tensors, int tensors_size, const T* expected_output_data,
+    T* output_data, int output_length, TfLiteFusedActivation activation,
+    float tolerance = 1e-5, int dilation_width = 1, int dilation_height = 1) {
```
Comment (njeffrie) [inline tensorflow/lite/micro/kernels/depthwise_conv_test.cc]: I'd would be nice to re-organize this test similarly to conv_test to take in a DepthwiseConvParams instead of passing dilation width and height directly.  In the future if we want to test multiple strides we can do so easily, and it keeps our tests consistent with one another.

**Review by njeffrie — CHANGES_REQUESTED** [review]
Thanks for your PR and patience merging it.  I have one slight preference for structuring depthwise_conv_test but otherwise it looks good.

**Comment by gbaned** [comment]
@giorgio-arenarm Could you please check reviewer comments and keep us posted. Thanks!

**Comment by giorgio-arenarm** [comment]
Thanks for the review, I have added TfLiteDepthwiseConvParams in the test's interface and rebased the commit :)

**Comment by giorgio-arenarm** [comment]
@petewarden  Gentle ping for review

**Review by njeffrie — DISMISSED** [review]
Sorry for the delay and thanks for making the changes!

**Review by petewarden — APPROVED** [review]

**Review by petewarden — DISMISSED** [review]
Approved, subject to confirmation that the test failures reported are unrelated to this change.

**Comment by gbaned** [comment]
@giorgio-arenarm Here are internal failures , can you please check.

'third_party/tensorflow/lite/micro/kernels/depthwise_conv_test.cc:887:18: error: no matching function for call to 'ValidateDepthwiseConvGoldens'
      kTfLiteOk, tflite::testing::ValidateDepthwiseConvGoldens(
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./third_party/tensorflow/lite/micro/testing/micro_test.h:112:17: note: expanded from macro 'TF_LITE_MICRO_EXPECT_EQ'
    if ((x) != (y)) {                                                          \
                ^
third_party/tensorflow/lite/micro/kernels/depthwise_conv_test.cc:30:14: note: candidate function template not viable: no known conversion from 'TfLiteFusedActivation' to 'TfLiteDepthwiseConvParams *' for 6th argument
TfLiteStatus ValidateDepthwiseConvGoldens(
             ^
third_party/tensorflow/lite/micro/kernels/depthwise_conv_test.cc:887:18: error: no matching function for call to 'ValidateDepthwiseConvGoldens'
      kTfLiteOk, tflite::testing::ValidateDepthwiseConvGoldens(
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./third_party/tensorflow/lite/micro/testing/micro_test.h:114:62: note: expanded from macro 'TF_LITE_MICRO_EXPECT_EQ'
                                   __FILE__, __LINE__, (x), (y));              \
                                                             ^
third_party/tensorflow/lite/micro/kernels/depthwise_conv_test.cc:30:14: note: candidate function template not viable: no known conversion from 'TfLiteFusedActivation' to 'TfLiteDepthwiseConvParams *' for 6th argument
TfLiteStatus ValidateDepthwiseConvGoldens(
             ^
2 errors generated.'

**Comment by giorgio-arenarm** [comment]
Hi @gbaned I've had a look at the error log but again it looks like maybe something is gone wrong while cherry picking the changes? `tensorflow/lite/micro/kernels/depthwise_conv_test.cc` file goes up to line 796 and the error seem to appear at line 887, moreover it complains about the 6th argument of `ValidateDepthwiseConvGoldens` being a `TfLiteFusedActivation` while throughout the file I checked that every time it's a `TfLiteDepthwiseConvParams *`. The code compiles on my side and the test pass. Any more info on the problem will be appreciated, thanks :)

**Comment by gbaned** [comment]
@petewarden Can you please assist with the above error ? Thanks!

**Comment by giorgio-arenarm** [comment]
Rebased. Could you please check whether the issue has been resolved?

**Review by njeffrie — DISMISSED** [review]

**Comment by advaitjain** [comment]
@giorgio-arenarm, apologies for the delayed response.

I just checked out your branch locally and am seeing the test fail with this command:
```
make -f tensorflow/lite/micro/tools/make/Makefile test_kernel_conv_test -j8
```

Error log:
```
Testing SimpleTestDilatedQuantized
expected_output_data[i] (1.8984370*2^7) near output_data[i] (1.3906247*2^7) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.6015621*2^7) near output_data[i] (1.1249999*2^7) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.1328123*2^7) near output_data[i] (1.312499*2^7) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.9921868*2^7) near output_data[i] (1.1406248*2^7) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.1484372*2^7) near output_data[i] (1.1796871*2^7) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.9921868*2^7) near output_data[i] (1.8906246*2^6) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
Testing SimpleTestQuantizedPerChannel
Testing SimpleTestDilatedQuantizedPerChannel
expected_output_data[i] (1.4374997*2^4) near output_data[i] (1.5624998*2^5) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.4999999*2^2) near output_data[i] (1.0*2^1) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.0*2^1) near output_data[i] (1.0*2^2) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.1249999*2^3) near output_data[i] (1.0*2^1) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
expected_output_data[i] (1.4999999*2^1) near output_data[i] (-1.7499998*2^2) failed at tensorflow/lite/micro/kernels/conv_test.cc:106
Testing SimpleTestQuantizedPerChannelRelu6
Testing Kernel1x1QuantizedPerChannel
Testing Kernel1x1QuantizedPerChannelRelu6
Testing FilterDimsNotMatchingAffineQuantization
tensorflow/lite/micro/kernels/conv.cc:238 affine_quantization->scale->size == 1 || affine_quantization->scale->size == filter->dims->data[kConvQuanti was not true.
tensorflow/lite/micro/kernels/conv.cc:238 affine_quantization->scale->size == 1 || affine_quantization->scale->size == filter->dims->data[kConvQuanti was not true.
Testing BroadcastPerLayerQuantizationToPerChannelShouldMatchGolden
9/11 tests passed
~~~SOME TESTS FAILED~~~
```

If you could take a look, that would be great.

**Comment by giorgio-arenarm** [comment]
Hi @advaitjain sorry but I haven't been able to reproduce the issue by checking out this branch and issuing the same command; everything passes for me. Could you please tell me what binary are your running after make?

**Comment by advaitjain** [comment]
I get the error with:
```
make -f tensorflow/lite/micro/tools/make/Makefile clean clean_downloads
make -f tensorflow/lite/micro/tools/make/Makefile test_kernel_conv_test -j8
```

and also if I run this binary:
```
tensorflow/lite/micro/tools/make/gen/linux_x86_64/bin/kernel_conv_test
```

The internal CI builds are also failing with this same error.

I also get an error with asan:
```
CC=clang BAZEL_COMPILER=llvm bazel run --copt=-DADDRESS_SANITIZER     --copt=-fsanitize=address --linkopt=-fsanitize=address tensorflow/lite/micro/kernels:conv_test_binary
```

Error log:
```
Testing SimpleTestDilatedQuantized
=================================================================
==44784==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffe456a810c at pc 0x0000005a5dd7 bp 0x7ffe456a4990 sp 0x7ffe456a4988
WRITE of size 1 at 0x7ffe456a810c thread T0
    #0 0x5a5dd6 in tflite::reference_ops::Conv(tflite::ConvParams const&, tflite::RuntimeShape const&, unsigned char const*, tflite::RuntimeShape const&, unsigned char const*, tflite::RuntimeShape const&, int const*, tflite::RuntimeShape const&, unsigned char*, tflite::RuntimeShape const&, unsigned char*, void*) (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x5a5dd6)
    #1 0x5a481a in tflite::ops::micro::conv::EvalQuantized(TfLiteContext*, TfLiteNode*, TfLiteConvParams*, tflite::ops::micro::conv::OpData*, TfLiteTensor const*, TfLiteTensor const*, TfLiteTensor const*, TfLiteTensor*, TfLiteTensor*, TfLiteTensor*) (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x5a481a)
    #2 0x5a9ce7 in tflite::ops::micro::conv::Eval(TfLiteContext*, TfLiteNode*) (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x5a9ce7)
    #3 0x4f8e46 in TfLiteStatus tflite::testing::(anonymous namespace)::ValidateConvGoldens<unsigned char>(TfLiteTensor*, int, unsigned char const*, unsigned char*, int, TfLiteConvParams*, float) (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x4f8e46)
    #4 0x4f6c94 in tflite::testing::(anonymous namespace)::TestConvQuantizedPerLayer(int const*, float const*, unsigned char*, float, int const*, float const*, unsigned char*, float, int const*, float const*, int*, int const*, float const*, unsigned char*, unsigned char*, float, TfLiteConvParams*) (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x4f6c94)
    #5 0x4f3a96 in main (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x4f3a96)
    #6 0x7fb7db481bba in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26bba)
    #7 0x41f1e9 in _start (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x41f1e9)

Address 0x7ffe456a810c is located in stack of thread T0 at offset 620 in frame
    #0 0x4f30bf in main (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x4f30bf)

  This frame has 115 object(s):
    [32, 40) 'error_reporter'
    [64, 112) 'output_data'
    [144, 152) 'output_data13'
    [176, 196) 'kFilterShape'
    [240, 272) 'filter_values'
    [304, 312) 'kBiasShape'
    [336, 340) 'bias_values'
    [352, 372) 'kOutputShape'
    [416, 424) 'expected_output'
    [448, 460) 'output_data38'
    [480, 496) 'input_quantized'
    [512, 524) 'filter_quantized'
    [544, 556) 'bias_quantized'
    [576, 588) 'golden_quantized'
    [608, 620) 'output_data61' <== Memory access at offset 620 overflows this variable
    [640, 660) 'input_shape'
    [704, 896) 'input_data'
    [960, 980) 'output_shape'
    [1024, 1120) 'golden_data'
    [1152, 1200) 'input_quantized65'
    [1232, 1244) 'filter_quantized66'
    [1264, 1276) 'bias_quantized67'
    [1296, 1320) 'golden_quantized68'
    [1360, 1384) 'conv_params'
    [1424, 1436) 'output_data95'
    [1456, 1472) 'input_quantized98'
    [1488, 1500) 'filter_quantized99'
    [1520, 1532) 'bias_quantized100'
    [1552, 1564) 'golden_quantized101'
    [1584, 1600) 'zero_points'
    [1616, 1632) 'scales'
    [1648, 1660) 'output_data126'
    [1680, 1700) 'input_shape132'
    [1744, 1936) 'input_data133'
    [2000, 2020) 'output_shape135'
    [2064, 2160) 'golden_data136'
    [2192, 2240) 'input_quantized137'
    [2272, 2284) 'filter_quantized138'
    [2304, 2316) 'bias_quantized139'
    [2336, 2360) 'golden_quantized140'
    [2400, 2416) 'zero_points141'
    [2432, 2448) 'scales142'
    [2464, 2488) 'conv_params143'
    [2528, 2540) 'output_data175'
    [2560, 2572) 'bias_values176'
    [2592, 2640) 'golden_data177'
    [2672, 2688) 'input_quantized182'
    [2704, 2716) 'filter_quantized183'
    [2736, 2748) 'bias_quantized184'
    [2768, 2780) 'golden_quantized185'
    [2800, 2816) 'zero_points186'
    [2832, 2848) 'scales187'
    [2864, 2888) 'conv_params213'
    [2928, 2948) 'kInputShape'
    [2992, 3056) 'kInputData'
    [3088, 3108) 'kFilterShape219'
    [3152, 3200) 'kFilterData'
    [3232, 3240) 'kBiasShape228'
    [3264, 3276) 'kBiasData'
    [3296, 3316) 'kOutputShape229'
    [3360, 3372) 'output_data230'
    [3392, 3440) 'kGoldenData'
    [3472, 3484) 'golden_quantized237'
    [3504, 3528) 'conv_params266'
    [3568, 3588) 'kInputShape267'
    [3632, 3696) 'kInputData276'
    [3728, 3748) 'kFilterShape277'
    [3792, 3840) 'kFilterData286'
    [3872, 3880) 'kBiasShape289'
    [3904, 3916) 'kBiasData290'
    [3936, 3956) 'kOutputShape291'
    [4000, 4012) 'output_data293'
    [4032, 4080) 'kGoldenData294'
    [4112, 4124) 'golden_quantized303'
    [4144, 4156) 'output_data333'
    [4176, 4192) 'input_quantized336'
    [4208, 4220) 'filter_quantized337'
    [4240, 4252) 'bias_quantized338'
    [4272, 4284) 'golden_quantized339'
    [4304, 4320) 'zero_points340'
    [4336, 4352) 'scales341'
    [4368, 4388) 'filter_zero_points'
    [4432, 4452) 'filter_scales'
    [4496, 4520) 'filter_quant'
    [4560, 4584) 'bias_quant'
    [4624, 4736) 'input_tensor'
    [4768, 4880) 'filter_tensor'
    [4912, 5024) 'bias_tensor'
    [5056, 5168) 'output_tensor'
    [5200, 5208) 'input_scales'
    [5232, 5240) 'input_zero_points'
    [5264, 5288) 'input_quant'
    [5328, 5776) 'tensors'
    [5840, 5852) 'output_data464'
    [5872, 5888) 'input_quantized468'
    [5904, 5916) 'filter_quantized469'
    [5936, 5948) 'bias_quantized470'
    [5968, 5980) 'golden_quantized471'
    [6000, 6112) 'input_tensor488'
    [6144, 6152) 'input_zero_points492'
    [6176, 6184) 'input_scales493'
    [6208, 6232) 'input_quant494'
    [6272, 6384) 'filter_tensor509'
    [6416, 6424) 'filter_zero_points513'
    [6448, 6456) 'filter_scales514'
    [6480, 6504) 'filter_quant515'
    [6544, 6656) 'bias_tensor532'
    [6688, 6696) 'bias_zero_points'
    [6720, 6728) 'bias_scales'
    [6752, 6776) 'bias_quant536'
    [6816, 6928) 'output_tensor551'
    [6960, 6968) 'output_zero_points'
    [6992, 7000) 'output_scales'
    [7024, 7048) 'output_quant'
    [7088, 7536) 'tensors572'
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow (/home/advaitjain/.cache/bazel/_bazel_advaitjain/d870524cf2b6c91d24e6d0f78ed0142b/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/micro/kernels/conv_test_binary+0x5a5dd6) in tflite::reference_ops::Conv(tflite::ConvParams const&, tflite::RuntimeShape const&, unsigned char const*, tflite::RuntimeShape const&, unsigned char const*, tflite::RuntimeShape const&, int const*, tflite::RuntimeShape const&, unsigned char*, tflite::RuntimeShape const&, unsigned char*, void*)
Shadow bytes around the buggy address:
  0x100048accfd0: 00 00 00 00 f1 f1 f1 f1 00 f2 f2 f2 f8 f8 f8 f8
  0x100048accfe0: f8 f8 f2 f2 f2 f2 f8 f2 f2 f2 f8 f8 f8 f2 f2 f2
  0x100048accff0: f2 f2 f8 f8 f8 f8 f2 f2 f2 f2 f8 f2 f2 f2 f8 f2
  0x100048acd000: f8 f8 f8 f2 f2 f2 f2 f2 f8 f2 f2 f2 f8 f8 f2 f2
  0x100048acd010: f8 f8 f2 f2 f8 f8 f2 f2 f8 f8 f2 f2 f8 f8 f2 f2
=>0x100048acd020: 00[04]f2 f2 00 00 04 f2 f2 f2 f2 f2 00 00 00 00
  0x100048acd030: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100048acd040: 00 00 00 00 f2 f2 f2 f2 f2 f2 f2 f2 00 00 04 f2
  0x100048acd050: f2 f2 f2 f2 00 00 00 00 00 00 00 00 00 00 00 00
  0x100048acd060: f2 f2 f2 f2 00 00 00 00 00 00 f2 f2 f2 f2 00 04
  0x100048acd070: f2 f2 00 04 f2 f2 00 00 00 f2 f2 f2 f2 f2 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==44784==ABORTING
```

**Comment by gbaned** [comment]
@giorgio-arenarm Can you please check @advaitjain's comments and keep us posted. Thanks!

**Comment by giorgio-arenarm** [comment]
Done. The bazel command revealed the error; I wasn't getting it because my machine was allocating extra memory even though the code was allocating less than needed. Could you please double check that this works now? Thanks

**Review by advaitjain — DISMISSED** [review]

**Comment by advaitjain** [comment]
@giorgio-arenarm, if possible please add new commits during the review process so that it is easy to see what the incremental changes are to the PR. Right now there is a single commit which makes it hard to follow along the review process.

**Comment by gbaned** [comment]
@giorgio-arenarm Can you please check @advaitjain's comments and keep us posted. Thanks!

**Comment by giorgio-arenarm** [comment]
@gbaned All required changes have been carried out. Thanks

**Comment by advaitjain** [comment]
@gbaned, this PR is ready to go. My comments were only for future PRs.

**Comment by advaitjain** [comment]
Confirming that conv_test passes the internal CI builds.

@giorgio-arenarm , I should have mentioned this in my earlier comment that depthwise_conv_test also fails with asan and will likely need a similar fix.

```
CC=clang BAZEL_COMPILER=llvm bazel run --copt=-DADDRESS_SANITIZER     --copt=-fsanitize=address --linkopt=-fsanitize=address tensorflow/lite/micro/kernels:depthwise_conv_test_binary
```

**Comment by giorgio-arenarm** [comment]
Hi @advaitjain I've had a look at the depthwise_conv failures and it doesn't seem to be the same error as conv. This time the amount of memory allocated for the vectors is right but the values are wrong. Since this patch is quite old, a lot might have changed in the implementation of the functions, so I just changed the golden values, matching what the reference kernels expect. Please let me know if this is ok, thanks!

**Comment by advaitjain** [comment]
Odd that the output values changed so significantly. It is what it is -- let's get this PR merged.

**Review by advaitjain — DISMISSED** [review]

**Comment by advaitjain** [comment]
sorry one more test failure, portable_optimized_depthwise_conv_test.

Let's make that work with asan too, just to be sure.

Hopefully after this we're done. Apologies for the long round trip times with the CI failures.

```
CC=clang BAZEL_COMPILER=llvm bazel run --copt=-DADDRESS_SANITIZER     --copt=-fsanitize=address --linkopt=-fsanitize=address tensorflow/lite/micro/kernels:portable_optimized_depthwise_conv_test_binary
```

**Comment by giorgio-arenarm** [comment]
Hi @advaitjain these errors are very interesting since they confirm my concern from before. Portable optimized depth_conv works fine with the golden values of the previous commit, which were the ones that were correct some versions ago. However, the regular depth_conv now fails with those values and I had to change them. In my opinion the regular depth_conv has had some changes recently that made it yield wrong result, but really either one could be the problem, it needs more investigation. Could you please double check what version I should go with? Thanks

**Comment by advaitjain** [comment]
@giorgio-arenarm, we'll need to look into this in more detail. I have an internal bug filed and will get back to you once we figure out what is happening here.

**Comment by gbaned** [comment]
@advaitjain Any update on this PR? Please. Thanks!

**Review by petewarden — DISMISSED** [review]

**Comment by rthadur** [comment]
@giorgio-arenarm seeing this error internally , can you please check 

`tensorflow/lite/micro/kernels/depthwise_conv_test.cc:1038:25: error: no matching function for call to 'ValidateDepthwiseConvGoldens'
  TfLiteStatus status = tflite::testing::ValidateDepthwiseConvGoldens(`

**Comment by advaitjain** [comment]
@giorgio-arenarm: thanks for your patience.

Please merge master one more time.

934df8d has removed the portable optimized kernels (thanks to the great work from you guys on the CMSIS-NN implementations) so you as long as the reference kernels pass, we should be good to go.

There may very well be an issue with the reference kernels, but we will tackle that separately.

**Review by rthadur — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.
